graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java

```diff
@@ -73,27 +88,51 @@ public SearchSourceBuilder create(SearchesConfig config) {
                     .numOfFragments(0);
             searchSourceBuilder.highlighter(highlightBuilder);
         }
+    }
 
-        return searchSourceBuilder;
+    private void applyPaginationIfPresent(SearchSourceBuilder searchSourceBuilder, SearchCommand command) {
+        command.offset().ifPresent(searchSourceBuilder::from);
+        command.limit().ifPresent(searchSourceBuilder::size);
     }
 
-    @Nullable
-    private QueryBuilder standardFilters(TimeRange range, String filter) {
-        BoolQueryBuilder bfb = null;
 
-        if (range != null) {
-            bfb = QueryBuilders.boolQuery()
-                    .must(TimeRangeQueryFactory.create(range));
-        }
+    private void applyStreamsFilter(BoolQueryBuilder filteredQueryBuilder, SearchCommand command) {
+        command.streams()
+                .map(this::buildStreamIdFilter)
+                .ifPresent(filteredQueryBuilder::filter);
+    }
+
+    private BoolQueryBuilder buildStreamIdFilter(Set<String> streams) {
```

Comment:
I think we should centralize this logic somewhere. Equivalent methods are also contained in `IndexToolsAdapterES6`, `indexToolsAdapterES7`, and `SearchesAdapterES6`.

Reply:
Same here. I would prefer to do this in a separate PR though, as it is non-trivial to make it usable across multiple storage driver modules.

Reply:
:+1: